### DISCUSSION TITLE
Change Button style based on enabled property

### DISCUSSION THREAD

**Discussion by cozimus** [beeware/toga#2361] [Q&A]
Hello everyone!
In your opinion, what is the best approach to change the style of a button (let say the background color) based on it's `enabled` property?
There is some CSS-like way, or is it better to just overwrite the default enabled.setter, or I'm missing some easy way😅?
At current time I'm simply using:
```
button.enabled = False
button.style.update(background_color=RED)
...
button.enabled = True
button.style.update(background_color=GREEN)
```
Every time.
Any suggestion? Thanks a lot!

**Comment by freakboy3742** [comment]
There's no built-in CSS-like mechanism to do this. A subclassed widget that overrides the `enabled` setter is about as good a solution as you're going to get.
